### ISSUE TITLE
Ambiguous Type Check in @langchain/aws

### ISSUE THREAD

**Issue by wailinkyaww** [langchain-ai/langchainjs#6101]
### Checked other resources

- [X] I added a very descriptive title to this issue.
- [X] I searched the LangChain.js documentation with the integrated search.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I am sure that this is a bug in LangChain.js rather than my code.
- [X] The bug is not resolved by updating to the latest stable version of LangChain (or the specific integration package).

### Example Code

This is from `@langchain/aws`.

Link to source code - https://github.com/langchain-ai/langchainjs/blob/dbb3b59623ee5ce4bf675bbf41a6e3e044ae0802/libs/langchain-aws/src/common.ts#L378

```js
if (contentBlockDelta.delta.text) {
    return new ChatGenerationChunk({
      text: contentBlockDelta.delta.text,
      message: new AIMessageChunk({
        content: contentBlockDelta.delta.text,
      }),
    });
}
```

### Error Message and Stack Trace (if applicable)

*Error Message*

TypeError: undefined is not an object (evaluating 'unsupportedField[0][0]')

*Stack Trace*

The error is coming from - https://github.com/langchain-ai/langchainjs/blob/dbb3b59623ee5ce4bf675bbf41a6e3e044ae0802/libs/langchain-aws/src/common.ts#L401

### Description

- I am trying to integrate ChatBedrockConverse from `@langchain/aws` with streaming.
- Sometimes, the language model produces `empty string` as a token which will cause `contentBlockDelta.delta.text` to be an empty string.
- As we are checking the `text` using truthy/falsy behaviour of JS, the chunk doesn't fall into the first if block.

This delta with an empty string ( text ) will goes into the else block.

https://github.com/langchain-ai/langchainjs/blob/dbb3b59623ee5ce4bf675bbf41a6e3e044ae0802/libs/langchain-aws/src/common.ts#L401

In line 401, we are again retrieving the `unsupportedField` by checking the existence of the value.
This will also fail because it is an empty string which will cause `unsupportedField` to be empty array.

Basically our error handler is throwing the error here.
 
---

### Resolution

I suggest we use the stricter type check instead of checking the existence of text from detla in `L370`.

```ts
// instead of
if (contentBlockDelta.delta.text) { }

// use string type check
if (typeof contentBlockDelta.delta.text === 'string') {}
```

I believe the library shouldn't care about what kind of tokens are passing through. As long as it is a valid `string`, we shouldn't care.

### System Info

Package => `@langchain-aws` - `^0.0.2`
Package manager => happens in both `yarn` and `bun` ( because it doesn't matter )

**Comment by bracesproul** [comment]
@wailinkyaww version `0.0.4` is now released with the fix!
